Remove mentions of `"reversible"` method (#2078)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Bug fixes</h3>

> * The available `diff_method`s for QNodes given in an error message and the
>   documentation have been corrected.
>   [(#2078)](https://github.com/PennyLaneAI/pennylane/pull/2078)

Reviewer: ```suggestion
* The available `diff_method` options for QNodes has been corrected in both the
  error messages and the documentation.
  [(#2078)](https://github.com/PennyLaneAI/pennylane/pull/2078)
```
